lib/solar_system.rb

```diff
@@ -0,0 +1,48 @@
+class SolarSystem
+  attr_reader :star_name, :planets # read only
+
+  # Each SolarSystem instance has a star and an array of Planets that orbit it.
+  def initialize(star_name)
+    @star_name = star_name
+    @planets = []
+  end
+
+  # Adds a Planet to this SolarSystem's planets array.
+  def add_planet(planet)
+    @planets << planet
+  end
+
+  # Returns a string listing all the Planets in this SolarSystem.
+  def list_planets
+    planet_list = "Planets orbiting #{@star_name}:\n"
+    @planets.each_with_index do |planet, i|
+      planet_list << "#{i+1}. #{planet.name}\n"
+    end
+    return planet_list
+  end
+
+  # Looks up a planet by name (case insensitive) and returns Planet's instance.
+  # If planet does not exist in this SolarSystem, raises ArgumentError.
+  def find_planet_by_name(planet_name)
+    found_planets = @planets.find_all do |planet|
+      planet.name.casecmp? planet_name
+    end
+    while found_planets.empty?
+      print "ERROR: #{planet_name} not found in this solar system.
+      Please enter another planet's name: "
+      planet_name = gets.chomp
```

Comment:
If you don't find the planet you should instead just return `nil`.  Let `main.rb` deal with the user.  Separate the concerns.